core/services/gateway/handler.dummy.go

```diff
@@ -0,0 +1,52 @@
+package gateway
+
+import (
+	"context"
+	"sync"
+)
+
+// DummyHandler forwards each request/response without doing any checks.
+type DummyHandler struct {
+	connMgr        DONConnectionManager
+	donConfig      *DONConfig
+	savedCallbacks map[string]UserCallback
+	mu             sync.Mutex
+}
+
+var _ Handler = (*DummyHandler)(nil)
+
+func (d *DummyHandler) Init(connMgr DONConnectionManager, donConfig *DONConfig) error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	d.connMgr = connMgr
+	d.donConfig = donConfig
+	d.savedCallbacks = make(map[string]UserCallback)
+	return nil
+}
+
+func (d *DummyHandler) HandleUserMessage(ctx context.Context, msg *Message, cb UserCallback) error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	d.savedCallbacks[msg.Body.MessageId] = cb
+	return d.connMgr.SendToAll(msg)
```

Comment:
Calling IO while holding the mutex. Probably fine for DummyHandler?

Reply:
Not really, fixing!